pkg/apis/core/helper/helpers.go

```diff
@@ -172,14 +172,11 @@ func IsDefaultNamespaceResource(name core.ResourceName) bool {
 		strings.Contains(string(name), core.ResourceDefaultNamespacePrefix)
 }
 
-var overcommitBlacklist = sets.NewString(string(core.ResourceNvidiaGPU))
-
 // IsOvercommitAllowed returns true if the resource is in the default
-// namespace and not blacklisted.
+// namespace and is not hugepages.
 func IsOvercommitAllowed(name core.ResourceName) bool {
```

Comment:
this is used in validation. this is used in the kubelet, which is allowed to skew two versions older than the apiserver. can we verify that a 1.10-level kubelet fails in a reasonable way if you specify the alpha resource on a pod with overcommit?

Reply:
If I understand correctly, this is used while validating the pod spec. So, from 1.11 (assuming this PR is merged), the API server will treat `alpha.kubernetes.io/nvidia-gpu` like any other `*kubernetes.io` prefixed resource (`IsDefaultNamespaceResource()`) and will allow pod specs with unequal requests and limits for this resource.

I would hope that when people upgrade their API server to 1.11, they won't submit any new pods using this resource.

But let's say we have a situation with API server running 1.11, kubelet running <1.11 with this alpha feature gate on, and the user submits a pod requesting this resource. In that case, the API server won't check whether requests=limits for this resource. And while assigning resources, kubelet will only look at the limits for this resource (like it does now).

Reply:
The only thing to keep in mind would be that currently the scheduler would schedule a pod requesting `*kubernetes.io/` prefixed resources to any node whether that node is exposing that resource or not (I don't know why this is the case).

Once we remove the special case for `alpha.kubernetes.io/nvidia-gpu`, this behavior will apply to `alpha.kubernetes.io/nvidia-gpu` as well. So, a pod requesting `alpha.kubernetes.io/nvidia-gpu` could be scheduled to any node.

Reply:
As @mindprince mentioned, the current scheduler behavior is to ignore resource request on non-existing resources in the kubernetes.io/ domain. This would cause the pod to be scheduled on a node without that requested resource. On the node, because Kubelet also runs GeneralPredicate, it would fail the pod during admission if it is running 1.10, which is actually the desired behavior for alpha.kubernetes.io/nvidia-gpu. However, if it is running 1.11, the pod would be started without proper gpu device setup.

@mindprince has initiated the discussion on whether we want to change this scheduler behavior on kubernetes.io/ domain resources in #50658 discussion. For now, I wonder whether we want to fail loudly during validation for resource request on alpha.kubernetes.io/nvidia-gpu to make sure that any users who haven't been aware of the deprecation of Accelerators feature can get the clear signal and move to the device plugin based solution. Then maybe after one or two releases, when #50658 is fully resolved, we can remove this special validation logic. Of course, Accelerators is an alpha feature, so it is debatable whether we want to add this special logic in resource validation code.

Reply:
> the current scheduler behavior is to ignore resource request on non-existing resources in the kubernetes.io/ domain.

that doesn't seem forward compatible, does it? if a new resource comes along and is requested by a pod, only kubelets that know about and declare they satisfy that resource should be running that pod, right?

> For now, I wonder whether we want to fail loudly during validation for resource request on alpha.kubernetes.io/nvidia-gpu to make sure that any users who haven't been aware of the deprecation of Accelerators feature can get the clear signal and move to the device plugin based solution. 

tightening validation brings a host of issues we want to avoid. it is better to let a pod in and it sit unscheduled than to prevent API writes because of stricter validation that can disrupt cleaning up the very resources that are newly considered invalid.

Reply:
Agree the desired behavior should be leaving the pod pending till the requested resource showing up, which is #50658 is about. I think @mindprince is working on a change to resolve  #50658 Scenario B. Agree it should be fine to leave the validation part out if both changes are merged in 1.11.

Reply:
https://github.com/kubernetes/kubernetes/pull/61860

Reply:
@liggitt This is addressed.